src/service/SettingsService.js

```diff
@@ -76,3 +76,11 @@ export const updateAllowNewMailAccounts = (allowed) => {
 	}
 	return axios.post(url, data).then((resp) => resp.data)
 }
+
+export const updateEnabledThreadSummary = async (enabled) => {
+	const url = generateUrl('/apps/mail/api/settings/threadsummary')
+	const data = {
+		enabled,
+	}
+	return await axios.post(url, data).then((resp) => resp.data)
```

Comment:
```suggestion
	const resp = await axios.post(url, data)
	return resp.data
```

Mixing async/await and callbacks is confusing.